src/Layouts/Partials/TransformPanel.vala

```diff
@@ -205,17 +218,54 @@ public class Akira.Layouts.Partials.TransformPanel : Gtk.Grid {
 
         attach (group_title (_("Opacity")), 0, 9, 3);
         attach (opacity_grid, 0, 10, 3);
+
+        event_bus.selected_items_changed.connect (on_selected_items_changed);
+    }
+
+    private void on_selected_items_changed (List<Lib.Models.CanvasItem> selected_items) {
+        if (selected_items.length () == 1) {
+            item = selected_items.nth_data (0);
+        } else {
+            if (item != null) {
+                item = null;
+            }
+        }
     }
 
     private void item_changed (Object object, ParamSpec spec) {
         //  debug ("item changed, param: %s", spec.name);
         update_fields ();
     }
 
+    private void reset_values () {
+        x.notify["value"].disconnect (x_notify_value);
+        y.notify["value"].disconnect (y_notify_value);
+        width.notify["value"].disconnect (width_notify_value);
+        height.notify["value"].disconnect (height_notify_value);
+        rotation.notify["value"].disconnect (rotation_notify_value);
+        opacity_adj.notify["value"].disconnect (opacity_notify_value);
+
+        x.value = 0.0;
+        y.value = 0.0;
+        width.value = 0.0;
+        height.value = 0.0;
+        opacity_adj.value = 100.0;
+        rotation.value = 0.0;
+        size_ratio = 1.0;
+        size_lock = false;
+
+        x.notify["value"].connect (x_notify_value);
+        y.notify["value"].connect (y_notify_value);
+        width.notify["value"].connect (width_notify_value);
+        height.notify["value"].connect (height_notify_value);
+        rotation.notify["value"].connect (rotation_notify_value);
+        opacity_adj.notify["value"].connect (opacity_notify_value);
+    }
```

Comment:
Is this disconnection and reconnection necessary?
if so, please write a comment to explain why. Maybe we can find a cleaner solution here.

Reply:
This code was already there, I just copied the logic present in order not to create infinite loops of setters and getters (I suppose). 

I think that there should (and can) be a better way to do this, maybe right inside the LinkedInput widget, with different methods like `reset`, `update`, `bind`, etc, in order to avoid to have to do it in every LinkedInputEntry. Maybe could we leave for a future and more specific PR on LinkedInput? I was facing other issues with them, so maybe we can take care of 'em plus this one alltogheter. 

I'll add a comment to summarize this Github comment.